samples/winsvc.app/main.lua

```diff
@@ -0,0 +1,210 @@
+
+  gSvcStatus.dwServiceType = winsvc.SERVICE_WIN32_OWN_PROCESS
+  gSvcStatus.dwServiceSpecificExitCode = 0
+
+  -- Report initial status to the SCM
+
+  ReportSvcStatus(winsvc.SERVICE_START_PENDING, winsvc.NO_ERROR, 3000)
+
+  -- Perform service-specific initialization and work.
+
+  SvcInit(args, context)
+end
+
+
+local function SvcInstall()
+  local svcPath, err = winsvcaux.GetModuleFileName()
+  if svcPath == nil then
+    print('Cannot install service, service path unobtainable', winsvcaux.GetErrorString(err))
+    return
+  end
+
+  -- Get a handle to the SCM database
+  local schSCManager, err = winsvc.OpenSCManager(nil, nil, winsvc.SC_MANAGER_ALL_ACCESS)
+  if schSCManager == nil then
+    print('OpenSCManager failed', winsvcaux.GetErrorString(err))
+    return
+  end
+
+  -- Create the Service
+  local schService, tagid, err = winsvc.CreateService(
+    schSCManager,
+    svcname,
+    svcname,
+    winsvc.SERVICE_ALL_ACCESS,
+    winsvc.SERVICE_WIN32_OWN_PROCESS,
+    winsvc.SERVICE_DEMAND_START,
+    winsvc.SERVICE_ERROR_NORMAL,
+    svcPath,
+    nil,
+    nil,
+    nil,
+    nil)
+
+  if schService == nil then
+    print('CreateService failed', winsvcaux.GetErrorString(err))
+    winsvc.CloseServiceHandle(schSCManager)
+    return
+  end
+
+  print('Service installed successfully')
+
+  winsvc.CloseServiceHandle(schService)
+  winsvc.CloseServiceHandle(schSCManager)
+
+end
+
+
+local function SvcDelete()
+  -- Get a handle to the SCM database
+  local schSCManager, err = winsvc.OpenSCManager(nil, nil, winsvc.SC_MANAGER_ALL_ACCESS)
+  if schSCManager == nil then
+    print('OpenSCManager failed', winsvcaux.GetErrorString(err))
+    return
+  end
+
+  -- Open the Service
+  local schService, err = winsvc.OpenService(
+    schSCManager,
+    svcname,
+    winsvc.DELETE)
+
+  if schService == nil then
+    print('OpenService failed', winsvcaux.GetErrorString(err))
+    winsvc.CloseServiceHandle(schSCManager)
+    return
+  end
+
+  -- Delete the Service
+  local schService = winsvc.OpenService(
+    schSCManager,
+    svcname,
+    winsvc.DELETE)
+
+  local delsuccess, err = winsvc.DeleteService(schService)
+  if not delsuccess then
+    print('DeleteService failed', winsvcaux.GetErrorString(err))
+  else
+    print('DeleteService succeeded')
+  end
+
+  winsvc.CloseServiceHandle(schService)
+  winsvc.CloseServiceHandle(schSCManager)
+
+end
+
+
+-- Main Code
+if args[1] == 'install' then
+  SvcInstall()
+  return
+elseif args[1] == 'delete' then
+  SvcDelete()
+  return
+end
+
+local DispatchTable = {}
+DispatchTable[svcname] = { SvcMain, SvcHandler };
+
+if not winsvc.SpawnServiceCtrlDispatcher(DispatchTable, function(success, err)
+  if success then
+    print('Service ControlDispatcher returned after threads exited ok')
+  else
+    print('Service ControlDispatcher returned eith err', winsvcaux.GetErrorString(err))
+  end
+end) then
+  SvcReportEvent('SpawnServiceCtrlDispatcher Succeeded')
+end
+
+uv.run('default')
```

Comment:
`uv.run()` is not provided by luvi.  It's part of the luvit sugar at https://github.com/luvit/luvit/blob/b9dad57aca1f3eca2662a33e8af8002c4900b5c5/app/modules/luvit.lua#L29-L39